### ISSUE TITLE
`__pydantic_extra__` type definition not inherited

### ISSUE THREAD

**Issue by KasperZutterman** [pydantic/pydantic#8744]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

When inheriting from a Model, the `ConfigDict(extra='allow')` is inherited, but the `__pydantic_extra__` type declaration isn't.
I would expect this to be inherited as well.

This is the same behaviour as this test: https://github.com/pydantic/pydantic/blob/28ed33f1b70d4b9d626e8c614aed545137c24bc8/tests/test_main.py#L2839-L2868

### Example Code

```Python
from pydantic import BaseModel, ConfigDict, SecretStr


class Model(BaseModel):
    model_config = ConfigDict(extra='allow')
    __pydantic_extra__: dict[str, SecretStr]

    y: int


class Child(Model):
    pass


m = Model(a='1', y=2)
print(m.__pydantic_extra__)  # {'a': SecretStr('**********')}
c = Child(a='1', y=2)
print(c.__pydantic_extra__)  # {'a': '1'}
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.6.1
        pydantic-core version: 2.16.2
          pydantic-core build: profile=release pgo=true
                 install path: /Users/kasperzutterman/.pyenv/versions/3.12.1/envs/pydantic-3.12/lib/python3.12/site-packages/pydantic
               python version: 3.12.1 (main, Feb  2 2024, 09:13:09) [Clang 15.0.0 (clang-1500.1.0.2.5)]
                     platform: macOS-14.2.1-arm64-arm-64bit
             related packages: typing_extensions-4.9.0
                       commit: unknown
```

**Comment by sydney-runkle** [comment]
@KasperZutterman,

Ah yes, you're correct. This is a bug. Would you be interested in submitting a PR with a fix? I'd be happy to review :)

**Comment by KasperZutterman** [comment]
Hi @sydney-runkle,
I'm unfortunately not acquainted with the Pydantic codebase. Could you point me in the right direction for the fix?

**Comment by sydney-runkle** [comment]
@KasperZutterman,

No worries. Good news! This was actually fixed in https://github.com/pydantic/pydantic/commit/8b90de86033fbdfea01f775e25ece8222e6a4d8f, so I'm going to mark this as resolved. This fix will be released with 2.7!

**Comment by sydney-runkle** [comment]
Marking this as resolved :)
